rc/filetype/zig.kak

```diff
@@ -36,18 +36,24 @@ provide-module zig %§
 add-highlighter shared/zig regions
 add-highlighter shared/zig/code default-region group
 
-add-highlighter shared/zig/doc_comment      region '///[^/]' '$'       fill documentation
-add-highlighter shared/zig/comment          region '//' '$'            fill comment
+add-highlighter shared/zig/doc_comment region '///[^/]' '$' fill documentation
+add-highlighter shared/zig/comment region '//' '$' fill comment
 
-# TODO: highlight escape sequences within strings
-add-highlighter shared/zig/string_double    region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/zig/string_single    region "'" (?<!\\)(\\\\)*' fill string
-add-highlighter shared/zig/string_multiline region '\\\\' '$'          fill string
+# strings and characters
+add-highlighter shared/zig/string region '"' (?<!\\)(\\\\)*" group
+add-highlighter shared/zig/string/ fill string
+add-highlighter shared/zig/string/ regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:meta
+
+add-highlighter shared/zig/character region "'" (?<!\\)(\\\\)*' group
+add-highlighter shared/zig/character/ fill string
+add-highlighter shared/zig/character/ regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:meta
```

Comment:
There's probably some clean way to duplicate this but I did find it.

Reply:
Maybe give the string highlighter a name, and then use ref here. Untested:
```
add-highlighter shared/zig/string/escapes regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:meta
add-highlighter shared/zig/character/escapes ref zig/string/escapes
```

Reply:
@krobelus good thought, but it doesn't seem to work unfortunately :/

Reply:
It works for me:

```diff
diff --git a/rc/filetype/zig.kak b/rc/filetype/zig.kak
index 157083fe..b0fc32d9 100644
--- a/rc/filetype/zig.kak
+++ b/rc/filetype/zig.kak
@@ -45 +45 @@ add-highlighter shared/zig/string/ fill string
-add-highlighter shared/zig/string/ regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:meta
+add-highlighter shared/zig/string/escape regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:attribute
@@ -49 +49 @@ add-highlighter shared/zig/character/ fill string
-add-highlighter shared/zig/character/ regex '(?:\\n|\\r|\\t|\\\\|\\''|\\"|\\x[0-9a-fA-F]{2}|\\u\{[0-9a-fA-F]+\})' 0:meta
+add-highlighter shared/zig/character/ ref zig/string/escape
```

(I changed "meta" to "attribute" for illustration since "string" and "meta" are the same, I don't know which face should actually be used.)

![image](https://user-images.githubusercontent.com/6853872/101261065-a0c4a100-3734-11eb-9c6f-06e72ab8061e.png)

Reply:
@krobelus I still can't get it to work, what version of kakoune are you using to test?

Reply:
I have based it on 586f79c30de2185a18f5f769e625184dd10fa40f, but it also works with Kakoune v2020.09.01:

I can launch a different Kakoune version and source the modified zig.kak (and no other rc file/config) like this:

```
kak -n -e 'source rc/filetype/zig.kak; source colors/default.kak; edit t/test.zig'
```